### PR TITLE
build(ci): add bundle size checker for improved infrastructure

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/ngxs/store/issues/748

## What is the new behavior?

image

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by splincode** [comment]
@markwhitfeld please authorize for me

https://github.com/siddharthkp/bundlesize

image

- Authorize bundlesize for status access, copy the token provided to Travis CI.
- Add this token as BUNDLESIZE_GITHUB_TOKEN as environment parameter in your CIs project settings.
- Trigger Travis CI

File: .circleci/config.yml [hunk 1]
```diff
@@ -143,6 +143,15 @@ jobs:
           command: yarn run test:types
       # - *persist_coverage // we don't run code coverage for integration tests
 
+  bundlesize:
+    <<: *job_defaults
+    steps:
+      - *attach_workspace
+      - run:
+          name: Test types
```
Comment (arturovt) [inline .circleci/config.yml]: `name: Check the size of NGXS bundle`? :D

File: .travis.yml [hunk 2]
```diff
@@ -25,3 +25,6 @@ script:
 
   # INTEGRATION TEST
   - yarn run test:types
+
+  # CHECK BUNDLE SIZE
+  - yarn run bundlesize
```
Comment (arturovt) [inline .travis.yml]: Just `yarn bundlesize`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -143,6 +143,15 @@ jobs:
...
+          command: yarn run bundlesize
```
Comment (arturovt) [inline .circleci/config.yml]: Just `yarn bundlesize`?

File: .travis.yml (hunk 2 above)
Comment (splincode) [reply .travis.yml]: copy paste ```$ yarn run test:types``` ?)

File: .circleci/config.yml (hunk 1 above)
Comment (splincode) [reply .circleci/config.yml]: Good catch

File: .travis.yml (hunk 2 above)
Comment (arturovt) [reply .travis.yml]: Just meant that `run` is not needed :D

Comment (splincode) [reply .travis.yml]: I can remove run everywhere

Comment (splincode) [reply .travis.yml]: fix

File: .circleci/config.yml (hunk 3 above)
Comment (splincode) [reply .circleci/config.yml]: fixed

**Comment by splincode** [comment]
@arturovt @markwhitfeld I fixed CI problem

**Comment by markwhitfeld** [comment]
@splincode I am rerunning the builds with the new token in the env settings.
What will be different?

**Comment by splincode** [comment]
@markwhitfeld great
image

**Comment by arturovt** [comment]
@splincode Great feature :tada: !!

**Pull request merged**